session7/src/com/itschool/homework/HomeRunChallenge7.java

```diff
@@ -0,0 +1,29 @@
+package com.itschool.homework;
+
+import java.util.Scanner;
+
+public class HomeRunChallenge7 {
+    public static void main(String[] args) {
+        HomeRunChallenge7 home = new HomeRunChallenge7();
+        home.challengeRun7();
+    }
+
+    public void challengeRun7() {
+        Scanner input = new Scanner(System.in);
+        int a = 20;
+        int b;
+        do {
+            System.out.println("Ghiceste numarul");
+            b = input.nextInt();
+            if (b > a) {
+                System.out.println("Mai ghiceste odata, numarul este prea mare");
+                b = input.nextInt();
+            } else if (b < a) {
+                System.out.println("Mai ghiceste odata, numarul este prea mic");
+                b = input.nextInt();
```

Comment:
there are some bugs running the program.. most of them because you are asking here and in the `if` branch again for the user's input. Here is my program trace:
```
Ghiceste numarul
7
Mai ghiceste odata, numarul este prea mic
2
Ghiceste numarul
29
Mai ghiceste odata, numarul este prea mare
20

Process finished with exit code 0
```
Removing the variable `b` reassignment from here and from line 20 + moving `System.out.println("Ghiceste numarul");` before the `do-while` should do the trick and get something like:
```
Ghiceste numarul
7
Mai ghiceste odata, numarul este prea mic
2
Mai ghiceste odata, numarul este prea mic
29
Mai ghiceste odata, numarul este prea mare
20
Ai ghicit numarul

Process finished with exit code 0
```